storage/netstore.go

```diff
@@ -193,6 +200,16 @@ func (n *NetStore) Get(ctx context.Context, mode chunk.ModeGet, req *Request) (c
 			if ok {
 				ch, err = n.RemoteFetch(ctx, req, fi)
 				if err != nil {
+					if n.recoveryCallback != nil {
+						n.recoveryCallback(ctx, ref)
+						time.Sleep(500 * time.Millisecond) // TODO: view what the ideal timeout is
+						ch, err = n.RemoteFetch(ctx, req, fi)
+						if err != nil {
+							// timeout and not found
+							return nil, err
+						}
+						return ch, nil
+					}
```

Comment:
it was my understanding that we would wait and retry a certain amount of times after giving up on getting the chunk.

this only will call the hook, wait 500 ms, and try again, right? if it fails then, we're done.

we can defer this to a another issue for later if you wish, although maybe if we make use of the monitor at some point it's not so important to deal with right now.

what do you think?